### PR TITLE
Fix issues identified by clippy

### PR DESCRIPTION
**Pull request by chandlore**
Remediates the following issues identified by clippy:

[const_static_lifetime](https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#const_static_lifetime)
[filter_next](https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#filter_next)
[single_match](https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#single_match)
[single_char_pattern](https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#single_char_pattern)
[option_map_or_none](https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#option_map_or_none)
[match_ref_pats](https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#match_ref_pats)
[needless_borrow](https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#needless_borrow)
[or_fun_call](https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#or_fun_call) 😉

### CODE REVIEW

File: ui/src/sandbox.rs [hunk 1]
```diff
@@ -235,7 +230,7 @@ impl Sandbox {
         let version_output = vec_to_str(output.stdout)?;
 
         let mut info: BTreeMap<String, String> = version_output.lines().skip(1).filter_map(|line| {
-            let mut pieces = line.splitn(2, ":").fuse();
+            let mut pieces = line.splitn(2, ':').fuse();
```
Comment (shepmaster) [inline ui/src/sandbox.rs]: I really should profile this sometime; I'm almost certain that the string form is faster than the character one, but that's based on gut feeling.

**Comment by shepmaster** [comment]
Thanks!

File: ui/src/sandbox.rs (hunk 1 above)
Comment (chandlore) [reply ui/src/sandbox.rs]: Looks like the string version is about 10-15% slower.  I ran both versions of this function through [criterion.rs](https://github.com/japaric/criterion.rs), these were the results:

```
double quote            time:   [1.4755 us 1.4763 us 1.4770 us]           
               
Found 9 outliers among 100 measurements (9.00%)
  3 (3.00%) low mild
  6 (6.00%) high mild

single quote            time:   [1.2746 us 1.2754 us 1.2763 us]                          

Found 7 outliers among 100 measurements (7.00%)
  2 (2.00%) high mild
  5 (5.00%) high severe
```

Comment (shepmaster) [reply ui/src/sandbox.rs]: huh. guess the clippy devs did the analysis before doing the lint ;-)

**Pull request merged**